packages/flutter/test/material/switch_test.dart

```diff
@@ -485,4 +487,46 @@ void main() {
     semanticsTester.dispose();
     SystemChannels.accessibility.setMockMessageHandler(null);
   });
+
+  testWidgets('Switch.adaptive', (WidgetTester tester) async {
+    bool value = false;
+
+    Widget buildFrame(TargetPlatform platform) {
+      return MaterialApp(
+        theme: ThemeData(platform: platform),
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Material(
+              child: Center(
+                child: Switch.adaptive(
+                  value: value,
+                  onChanged: (bool newValue) {
+                    setState(() {
+                      value = newValue;
+                    });
+                  },
+                ),
+              ),
+            );
+          },
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildFrame(TargetPlatform.iOS));
+    expect(find.byType(CupertinoSwitch), findsOneWidget);
+
+    expect(value, isFalse);
+    await tester.tap(find.byType(Switch));
+    expect(value, isTrue);
+
+    await tester.pumpWidget(buildFrame(TargetPlatform.android));
+    await tester.pumpAndSettle(); // Finish the theme change animation.
```

Comment:
use pump(const Duration(milliseconds: 500)) or some such to make it deterministic?

Reply:
I meant that this will assert that it will transition at some point but we don't know when it started and when it ended. 

I'm just quoting the API doc :)

```
In general, it is better practice to figure out exactly why each frame is
needed, and then to [pump] exactly as many frames as necessary. This will
help catch regressions where, for instance, an animation is being started
one frame later than it should.
```

Reply:
I'm not testing the Theme animation here, just waiting for it to end.